web/src/app/dialogs/new-inspection/components/file-upload.component.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommonModule } from '@angular/common';
+import {
+  Component,
+  ElementRef,
+  inject,
+  input,
+  signal,
+  ViewChild,
+} from '@angular/core';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatSnackBar, MatSnackBarModule } from '@angular/material/snack-bar';
+import {
+  FILE_UPLOADER,
+  UplaodToken,
+  UploadStatus,
+} from './service/file-uploader';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+
+@Component({
+  selector: 'khi-new-inspection-file-upload',
+  templateUrl: './file-upload.component.html',
+  styleUrls: ['./file-upload.component.sass'],
+  imports: [
+    CommonModule,
+    MatFormFieldModule,
+    ReactiveFormsModule,
+    MatIconModule,
+    MatButtonModule,
+    MatSnackBarModule,
+    MatProgressSpinnerModule,
+  ],
+})
+export class FileUploadComponent {
+  /**
+   * The label of this upload form field.
+   */
+  label = input.required<string>();
+
+  /**
+   * The token provided from backend to request file.
+   */
+  uploadToken = input.required<UplaodToken>();
+
+  /**
+   * The description of this upload form field.
+   */
+  description = input('');
+
+  /**
+   * The error message returned from the backend about this field.
+   */
+  errorMessage = input('');
+
+  /**
+   * The status of upload for this file form.
+   */
+  uploadStatus = input(UploadStatus.Waiting);
```

Comment:
It's more like a question and maybe I miss something but when it's updated to Uploading or Verifying or Done? What is the advantage to have `uploaded` seperately?

Reply:
I can add detailed explain in this comment, but your question would be benefitial for the other code readers as well.
I added comment for the variable instead.

https://github.com/GoogleCloudPlatform/khi/blob/cd4e9021163c3d9c8fca465088e6c51cc8fa8081/web/src/app/dialogs/new-inspection/components/file-upload.component.ts#L79-L81

Let me know if it solves your question.